src/main/java/seedu/address/logic/commands/AddTagsCommand.java

```diff
@@ -0,0 +1,104 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.ReadOnlyPerson;
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+import seedu.address.model.person.exceptions.PersonNotFoundException;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Adds tags to an existing person in the address book.
+ */
+public class AddTagsCommand extends UndoableCommand {
+
+    public static final String COMMAND_WORD = "addtags";
+    public static final String COMMAND_ALIAS = "at";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Add tags to the person identified "
+            + "by the index number used in the last person listing. \n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "TAG [TAG] (can add 1 or more tags)... \n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + "NUS CS2103 classmate";
+
+    public static final String MESSAGE_ADD_TAGS_SUCCESS = "Added Tag/s to Person: %1$s";
+    public static final String MESSAGE_NO_TAG = "One or more tags must be added.";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
+
+    private final Index index;
+    private final Set<Tag> tags;
+
+    public AddTagsCommand(Index index, Set<Tag> tags) {
+        requireNonNull(index);
+        requireNonNull(tags);
+
+        this.index = index;
+        this.tags = tags;
+    }
+
+    @Override
+    protected CommandResult executeUndoableCommand() throws CommandException {
+        List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        if (tags.isEmpty()) {
+            throw new CommandException(MESSAGE_NO_TAG);
+        }
+
+        ReadOnlyPerson personToEdit = lastShownList.get(index.getZeroBased());
+        Person editedPerson = createEditedPerson(personToEdit, tags);
+
+        try {
+            model.updatePerson(personToEdit, editedPerson);
+        } catch (DuplicatePersonException dpe) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
```

Comment:
Well, this thing will never happen right?

I think `assert(false)` or just throw empty CommandException is better. The existence of `MESSAGE_DUPLICATE_PERSON` is, I feel, quite misleading.

Reply:
Yeah actually it could never happen, but CommandException's arg cannot be empty.

Reply:
Can we just pass an empty string or something?

Reply:
Hmm yeah can pass empty string, but in this case i think maybe assert(false) will be more indicative that it should never happen